sphinx/project.py

```diff
@@ -55,7 +56,13 @@ def discover(self, exclude_paths=[]):
         for filename in get_matching_files(self.srcdir, excludes):  # type: ignore
             docname = self.path2doc(filename)
             if docname:
-                if os.access(os.path.join(self.srcdir, filename), os.R_OK):
+                if docname in self.docnames:
+                    pattern = os.path.join(self.srcdir, docname) + '.*'
+                    files = [relpath(f, self.srcdir) for f in glob(pattern)]
+                    logger.warning(__('multiple files found for the document "%s": %r\n'
+                                      'Use %r for the build.'),
+                                   docname, files, self.doc2path(docname), once=True)
```

Comment:
Now `project.discover()` is called twice. So I needed to improve `logger.warning()` to suppress this warning on the　second call.

Reply:
Looks good.